src/ui/public/courier/fetch/call_client.js

```diff
@@ -76,17 +82,30 @@ export function CallClientProvider(Private, Promise, esAdmin, es) {
       });
     });
 
-
     // Now that all of THAT^^^ is out of the way, lets actually
     // call out to elasticsearch
-    Promise.map(executable, function (req) {
-      return Promise.try(req.getFetchParams, void 0, req)
-      .then(function (fetchParams) {
-        return (req.fetchParams = fetchParams);
+    promiseMapSettled(
+      executable,
+      function (request) {
+        return Promise.try(request.getFetchParams, void 0, request)
+        .then(function (fetchParams) {
+          return (request.fetchParams = fetchParams);
+        });
+      },
+      Promise)
+    .then(function (results) {
+      const requestsWithFetchParams = results.map((result, index) => {
```

Comment:
I'm not in love with this solution, but I've spent a couple hours trying to come up with a better one and none of my ideas will take less than a couple days... This works, and the scope of the change is impressively small, so I'm happy with it 😎